### PR TITLE
add license information, update README

### LINKED ISSUE #38: Include a terms of use paragraph in README?
Greetings all,

This seems like a wonderful resource to get Intel compilers working for open source development. I'd like to suggest adding a paragraph regarding the terms of use. You currently point to a CC-BY-4.0 and MIT license under the license heading, but this is insufficiently clear to me. Recalling the adage of:

> If something seems too good to be true, it probably is.

So I wasn't sure, and I checked this page:
https://software.intel.com/content/www/us/en/develop/articles/qualify-for-free-software.html

And a recent EULA:
https://software.intel.com/content/dam/develop/public/us/en/documents/intel-end-user-license-agreement-for-developer-tools-oct-2020.pdf

Which leads me to the following impressions:

* I can use excerpts from this repository to configure CI for open source software (public domain in this specific case), provided I attribute this repo
* I am allowed to install ifort on the CI machine
* I am allowed to use ifort to build a binary on the CI machine
* I am allowed to distribute binaries that were built on that CI machine

Is my understanding correct?

If my understanding is correct, then I'd suggest adding something like such a specifically worded list in the README, as well as providing a link to the EULA proper of course (if that also pleases the lawyerly types sufficiently).

### PR DESCRIPTION
**Pull request by rscohn2**
Resolves #38. I wrote something simple. We can get legal review offline.

Some updates to the README. The (R) is now properly superscripted. Unfortunately, you can't do superscripts in substitutions so I cut/paste a (R) symbol into the file.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -56,6 +59,13 @@ Azure    |c|       |c|       |c|          |c|   |c|     |c|
 ======== ========= ========= ============ ===== ======= ===========
 
 
+Cacheing
```
Comment (mmzakhar) [inline README.rst:62]: Should it be 'caching' w/o 'e' here?

File: README.rst [hunk 2]
```diff
@@ -56,6 +59,13 @@ Azure    |c|       |c|       |c|          |c|   |c|     |c|
...
+========
+
+Some of the oneAPI components are large and can take a few minutes to
+download install. To accelerate install, we demonstrate the use of
+cacheing.
```
Comment (mmzakhar) [inline README.rst:67]: and here?

**Review by mmzakhar — APPROVED** [review]

File: README.rst (hunk 1 above)
Comment (rscohn2) [reply README.rst:62]: Fixed

File: README.rst (hunk 2 above)
Comment (rscohn2) [reply README.rst:67]: Fixed

**Pull request merged**